### ISSUE TITLE
Avocado crashed: NetworkError: Name lookup failed for fedora20.vm

### ISSUE THREAD

**Issue by ruda** [avocado-framework/avocado#580]
Avocado should gives a gentle error message and do not crash when the hostname provide could not be resolved.
```
./scripts/avocado run --remote-hostname fedora20.vm --remote-username fedora --remote-password fedora passtest
LOGIN      : fedora@fedora20.vm:22
Avocado crashed: NetworkError: Name lookup failed for fedora20.vm
Traceback (most recent call last):

  File "/home/rmoura/Work/avocado-remote/avocado/job.py", line 310, in _run
    failures = self.test_runner.run_suite(test_suite)

  File "/home/rmoura/Work/avocado-remote/avocado/remote/runner.py", line 80, in run_suite
    self.result.setup()

  File "/home/rmoura/Work/avocado-remote/avocado/remote/result.py", line 87, in setup
    self._copy_tests()

  File "/home/rmoura/Work/avocado-remote/avocado/remote/result.py", line 54, in _copy_tests
    self.remote.makedir(self.remote_test_dir)

  File "/home/rmoura/Work/avocado-remote/avocado/utils/remote.py", line 133, in makedir
    self.run('mkdir -p %s' % remote_path)

  File "/home/rmoura/Work/avocado-remote/avocado/utils/remote.py", line 100, in run
    timeout=timeout)

  File "/usr/lib/python2.7/site-packages/fabric/network.py", line 639, in host_prompting_wrapper
    return func(*args, **kwargs)

  File "/usr/lib/python2.7/site-packages/fabric/operations.py", line 1042, in run
    shell_escape=shell_escape)

  File "/usr/lib/python2.7/site-packages/fabric/operations.py", line 909, in _run_command
    channel=default_channel(), command=wrapped_command, pty=pty,

  File "/usr/lib/python2.7/site-packages/fabric/state.py", line 390, in default_channel
    chan = _open_session()

  File "/usr/lib/python2.7/site-packages/fabric/state.py", line 382, in _open_session
    return connections[env.host_string].get_transport().open_session()

  File "/usr/lib/python2.7/site-packages/fabric/network.py", line 151, in __getitem__
    self.connect(key)

  File "/usr/lib/python2.7/site-packages/fabric/network.py", line 143, in connect
    self[key] = connect(user, host, port, cache=self)

  File "/usr/lib/python2.7/site-packages/fabric/network.py", line 533, in connect
    raise NetworkError('Name lookup failed for %s' % host, e)

NetworkError: Name lookup failed for fedora20.vm

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```

**Comment by clebergnu** [comment]
Agreed.

**Comment by adereis** [comment]
I reproduced it.

Is there a card for it? If not, can you please create one?

(IMO, we should prefer to report issues in trello and leave this for supporting users)

**Comment by ruda** [comment]
@adereis this was fixed yesterday in PR #659 and the card on Trello is here https://trello.com/c/mJZ6HE1v/408-bug-when-using-the-remote-vm-feature-if-the-setup-fails-it-will-dump-the-traceback-and-give-the-information-that-avocado-crashes
